src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -7252,13 +7252,13 @@ final boolean complete(LogonCommand logonCommand, TDSReader tdsReader) throws SQ
 
         LogonProcessor logonProcessor = new LogonProcessor(authentication);
         TDSReader tdsReader;
+        sessionRecovery.setConnectionRecoveryPossible(false);
         do {
             tdsReader = logonCommand.startResponse();
-            sessionRecovery.setConnectionRecoveryPossible(false);
             TDSParser.parse(tdsReader, logonProcessor);
         } while (!logonProcessor.complete(logonCommand, tdsReader));
 
-        if (sessionRecovery.isReconnectRunning() && !sessionRecovery.isConnectionRecoveryPossible()) {
+        if (sessionRecovery.isReconnectRunning() && !sessionRecovery.isConnectionRecoveryPossible() && routingInfo == null) {
```

Comment:
Formatting rules will wrap this. If you haven't yet, you should import the [project formatting rules](https://github.com/microsoft/mssql-jdbc/blob/main/Coding_Guidelines.md#1-general) file to your Eclipse settings.
![image](https://github.com/user-attachments/assets/995fe0f1-8c94-4a22-8ca1-24d0bb77b5e0)

You can then have Eclipse format on save so that formatting happens for you automatically:
![image](https://github.com/user-attachments/assets/b6c6056e-8b77-4749-9e05-5dd022ee581e)

```suggestion
        if (sessionRecovery.isReconnectRunning() && !sessionRecovery.isConnectionRecoveryPossible()
                && routingInfo == null) {
```

Reply:
Thanks for sharing.